operator/src/main/resources/scripts/auxImage.sh

```diff
@@ -32,8 +32,17 @@ UNKNOWN_SHELL=true
 
 checkEnv AUXILIARY_IMAGE_TARGET_PATH AUXILIARY_IMAGE_CONTAINER_NAME || exit 1
 
-initAuxiliaryImage > /tmp/auxiliaryImage.out 2>&1
-cat /tmp/auxiliaryImage.out
-mkdir -p ${AUXILIARY_IMAGE_TARGET_PATH}/auxiliaryImageLogs
-cp /tmp/auxiliaryImage.out ${AUXILIARY_IMAGE_TARGET_PATH}/auxiliaryImageLogs/${AUXILIARY_IMAGE_CONTAINER_NAME}.out
+if [[ "$AUXILIARY_IMAGE_CONTAINER_NAME" == "operator-aux-container"* ]]; then
+  initAuxiliaryImage > /tmp/auxiliaryImage.out 2>&1
+  cat /tmp/auxiliaryImage.out
+
+  mkdir -p ${AUXILIARY_IMAGE_TARGET_PATH}/auxiliaryImageLogs
+  cp /tmp/auxiliaryImage.out ${AUXILIARY_IMAGE_TARGET_PATH}/auxiliaryImageLogs/${AUXILIARY_IMAGE_CONTAINER_NAME}.out
+elif [[ "$AUXILIARY_IMAGE_CONTAINER_NAME" == "compatibility-mode-operator-aux-container"* ]]; then
+  initCompatibilityModeInitContainersWithLegacyAuxImages > /tmp/compatibilityModeInitContainers.out 2>&1
+  cat /tmp/compatibilityModeInitContainers.out
+
+  mkdir -p ${AUXILIARY_IMAGE_TARGET_PATH}/compatibilityModeInitContainerLogs
+  cp /tmp/compatibilityModeInitContainers.out ${AUXILIARY_IMAGE_TARGET_PATH}/compatibilityModeInitContainerLogs/${AUXILIARY_IMAGE_CONTAINER_NAME}.out
```

Comment:
For code clarity, maybe have a different file, 'auxImageV2.sh' for new approach, and leave 'auxImage.sh' unchanged.  For safety, IMO, it should remain unchanged regardless for 'old' pods...

Reply:
I have tested the current scripts and they work with both the old and new pods. I can create a different 'auxImageV2.sh' for the code clarity as you suggested.

Reply:
As suggested, I have added an 'else' to fail if the container name is unexpeted.

Reply:
I continue to prefer having two versions of this script.  One for "normal" internal use, and one for customer use. This way a customer that is using the custom path will not need to care about the container name, and cannot get it wrong...  Or, if the container name must have a certain format, then this way it will always (properly) expect the "compat" format instead of allowing for the (possibly incorrect) normal format...